Compiler warnings on `-Wstringop-overread`

From release verification on condaforge/mambaforge:

```
In file included from /arrow-nanoarrow/extensions/nanoarrow_ipc/../../src/nanoarrow/array_inline.h:27,
                 from /arrow-nanoarrow/extensions/nanoarrow_ipc/../../src/nanoarrow/nanoarrow.h:1100,
                 from /arrow-nanoarrow/extensions/nanoarrow_ipc/src/nanoarrow/nanoarrow_ipc.h:21,
                 from /arrow-nanoarrow/extensions/nanoarrow_ipc/src/nanoarrow/nanoarrow_ipc_reader_test.cc:22:
In function 'void ArrowBufferAppendUnsafe(ArrowBuffer*, const void*, int64_t)',
    inlined from 'ArrowErrorCode ArrowBufferAppend(ArrowBuffer*, const void*, int64_t)' at /arrow-nanoarrow/extensions/nanoarrow_ipc/../../src/nanoarrow/buffer_inline.h:124:26,
    inlined from 'virtual void NanoarrowIpcReader_StreamReaderExpectedSchema_Test::TestBody()' at /arrow-nanoarrow/extensions/nanoarrow_ipc/src/nanoarrow/nanoarrow_ipc_reader_test.cc:321:3:
/arrow-nanoarrow/extensions/nanoarrow_ipc/../../src/nanoarrow/buffer_inline.h:115:11: warning: 'void* memcpy(void*, const void*, size_t)' reading 280 bytes from a region of size 160 [-Wstringop-overread]
  115 |     memcpy(buffer->data + buffer->size_bytes, data, size_bytes);
      |     ~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/arrow-nanoarrow/extensions/nanoarrow_ipc/src/nanoarrow/nanoarrow_ipc_reader_test.cc: In member function 'virtual void NanoarrowIpcReader_StreamReaderExpectedSchema_Test::TestBody()':
/arrow-nanoarrow/extensions/nanoarrow_ipc/src/nanoarrow/nanoarrow_ipc_reader_test.cc:46:16: note: source object 'kSimpleRecordBatch' of size 160
   46 | static uint8_t kSimpleRecordBatch[] = {
      |                ^~~~~~~~~~~~~~~~~~
```

--- a/extensions/nanoarrow_ipc/src/nanoarrow/nanoarrow_ipc_reader_test.cc
+++ b/extensions/nanoarrow_ipc/src/nanoarrow/nanoarrow_ipc_reader_test.cc
@@ -318,7 +318,7 @@ TEST(NanoarrowIpcReader, StreamReaderExpectedRecordBatch) {
 TEST(NanoarrowIpcReader, StreamReaderExpectedSchema) {
   struct ArrowBuffer input_buffer;
   ArrowBufferInit(&input_buffer);
-  ASSERT_EQ(ArrowBufferAppend(&input_buffer, kSimpleRecordBatch, sizeof(kSimpleSchema)),
+  ASSERT_EQ(ArrowBufferAppend(&input_buffer, kSimpleRecordBatch, sizeof(kSimpleRecordBatch)),
             NANOARROW_OK);
 
   struct ArrowIpcInputStream input;

--- a/extensions/nanoarrow_ipc/src/nanoarrow/nanoarrow_ipc_reader_test.cc
+++ b/extensions/nanoarrow_ipc/src/nanoarrow/nanoarrow_ipc_reader_test.cc
@@ -318,8 +318,9 @@ TEST(NanoarrowIpcReader, StreamReaderExpectedRecordBatch) {
 TEST(NanoarrowIpcReader, StreamReaderExpectedSchema) {
   struct ArrowBuffer input_buffer;
   ArrowBufferInit(&input_buffer);
-  ASSERT_EQ(ArrowBufferAppend(&input_buffer, kSimpleRecordBatch, sizeof(kSimpleRecordBatch)),
-            NANOARROW_OK);
+  ASSERT_EQ(
+      ArrowBufferAppend(&input_buffer, kSimpleRecordBatch, sizeof(kSimpleRecordBatch)),
+      NANOARROW_OK);
 
   struct ArrowIpcInputStream input;
   ASSERT_EQ(ArrowIpcInputStreamInitBuffer(&input, &input_buffer), NANOARROW_OK);

